projects/ng-live-docs/src/compodoc/compodoc-retriever.service.ts

```diff
@@ -55,11 +55,45 @@ export class CompoDocRetrieverService implements DocumentationRetrieverService {
 
     public getInputParameters(component: Type<unknown>): ApiParameters[] {
         const comp = this.getComponent(component);
-        return comp.inputsClass || [];
+        return comp.inputsClass ? comp.inputsClass.map((input) => {
+            return {...input, typeLink: this.getTypeLink(input.type)};
+        }) : [];
     }
 
     public getOutputParameters(component: Type<unknown>): ApiParameters[] {
         const comp = this.getComponent(component);
-        return comp.outputsClass || [];
+        return comp.outputsClass ? comp.outputsClass.map((output) => {
+            return {...output, typeLink: this.getTypeLink(output.type)};
+        }) : [];
+    }
+
+    private getTypeLink(type: string): string {
+        // Extract the outermost Typescript type of the given type string.
+        const rawType = type.split('[')[0].split('<')[0];
+        // Check every documentation JSON registered with the library.
+        for (const documentationJson of this.documentationJson) {
+            // Check every documentation type in the given documentation
+            for (const typeLink of Object.keys(documentationJson)) {
+                // If it is a list, attempt to find the given raw type
+                if (documentationJson[typeLink].find) {
+                    const typeDoc = documentationJson[typeLink].find(c => c.name === rawType);
+                    if (typeDoc) {
+                        return typeLink + '/' + rawType + '.html';
+                    }
+                } else {
+                    // Else check one level deeper because it is the 'miscellaneous' section.
+                    for (const innerTypeLink of Object.keys(documentationJson[typeLink])) {
+                        // If it is an array, find the given raw type.
+                        if (documentationJson[typeLink][innerTypeLink].find) {
```

Comment:
I still don't understand what it going on here, we may have to pair to figure out how to say something meaningful so someone doesn't have to debug this before they can understand it. Maybe using type guards could help. Maybe an inner function that knows how to extract info from each node being traversed... Let's chat about this...